### ISSUE TITLE
BUG: Missing overloads for `np.number`-derived ABCs

### ISSUE THREAD

**Issue by Jasha10** [numpy/numpy#20442]
### Describe the issue:

Why operating on arrays typed as `npt.NDArray[np.number[Any]]`, the mypy plugin reports that `ndarray.__mul__` cannot accept argument type `int`.

I would expect that an array of `np.number` can be multiplied by an integer.

### Reproduce the code example:

```python
# tmp.py
from typing import Any

import numpy as np
from numpy import typing as npt

arr: npt.NDArray[np.number[Any]] = np.array([1, 2])
arr * 3  # mypy error
```

Running `mypy` on the above code gives an error.

### Error message:

```text
$ mypy tmp.py
tmp.py:8: error: No overload variant of "__mul__" of "ndarray" matches argument type "int"
tmp.py:8: note: Possible overload variants:
tmp.py:8: note:     def __mul__(self, Union[str, bytes, Sequence[Union[str, bytes]], Sequence[Sequence[Union[str, bytes]]], Sequence[Sequence[Sequence[Union[str, bytes]]]], Sequence[Sequence[Sequence[Sequence[Union[str, bytes]]]]]]) -> NoReturn
tmp.py:8: note:     def __mul__(self, Union[_SupportsArray[dtype[object_]], Sequence[_SupportsArray[dtype[object_]]], Sequence[Sequence[_SupportsArray[dtype[object_]]]], Sequence[Sequence[Sequence[_SupportsArray[dtype[object_]]]]], Sequence[Sequence[Sequence[Sequence[_SupportsArray[dtype[object_]]]]]]]) -> Any
tmp.py:8: note:     def __mul__(self, Sequence[Sequence[Sequence[Sequence[Sequence[Any]]]]]) -> Any
Found 1 error in 1 file (checked 1 source file)
```

### NumPy/Python version information:

```text
$ mypy --version
mypy 0.910
$ python
Python 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:23:11)
[GCC 9.4.0] on linux
...
>>> import numpy; print(numpy.__version__)
1.21.4

**Comment by BvB93** [comment]
> I would expect that an array of `np.number` can be multiplied by an integer.

As of the moment dtype-support for the `np.inexact`- and `np.number`-ABCs isn't all that widespread, so for the time being it might be simpler to stick to `np.integer[Any] | np.floating[Any]`.

I agree this is something that should be fixed though, but with the branching of the 1.22 branch the changes will have to wait for numpy 1.23.

**Comment by Jasha10** [comment]
I can no longer reproduce this issue with numpy v1.23.1 and mypy v0.961.
